tests/core/test_inputs.c

```diff
@@ -14,39 +15,103 @@ static byte_t tx_id1[TRANSACTION_ID_BYTES] = {255, 255, 255, 255, 255, 255, 255,
 static byte_t tx_id2[TRANSACTION_ID_BYTES] = {126, 127, 95,  249, 151, 44,  243, 150, 40,  39, 46,
                                               190, 54,  49,  73,  171, 165, 88,  139, 221, 25, 199,
                                               90,  172, 252, 142, 91,  179, 113, 2,   177, 58};
+static byte_t tx_id3[TRANSACTION_ID_BYTES] = {30,  49,  142, 249, 151, 44,  243, 150, 40,  39, 46,
+                                              190, 54,  200, 73,  171, 165, 88,  139, 221, 25, 199,
+                                              90,  172, 252, 142, 91,  179, 113, 120, 110, 70};
 
 void setUp(void) {}
 
 void tearDown(void) {}
 
 void test_utxo_input() {
-  utxo_input_ht* inputs = utxo_inputs_new();
+  utxo_inputs_list_t* inputs = utxo_inputs_new();
   TEST_ASSERT_NULL(inputs);
 
-  TEST_ASSERT_EQUAL_UINT16(0, utxo_inputs_count(&inputs));
-  TEST_ASSERT(utxo_inputs_add(&inputs, tx_id0, UINT8_MAX) == -1);
-  TEST_ASSERT_EQUAL_UINT16(0, utxo_inputs_count(&inputs));
-  TEST_ASSERT(utxo_inputs_add(&inputs, tx_id0, 0) == 0);
-  TEST_ASSERT_EQUAL_UINT16(1, utxo_inputs_count(&inputs));
+  // print out an empty list
+  utxo_inputs_print(inputs, 0);
 
-  // transaction ID doesn't exist.
-  TEST_ASSERT_NULL(utxo_inputs_find_by_id(&inputs, tx_id2));
+  // get count of empty list
+  TEST_ASSERT_EQUAL_UINT16(0, utxo_inputs_count(inputs));
 
-  // add more tx IDs
-  TEST_ASSERT(utxo_inputs_add(&inputs, tx_id1, 1) == 0);
-  TEST_ASSERT(utxo_inputs_add(&inputs, tx_id2, 2) == 0);
-  TEST_ASSERT_EQUAL_UINT16(3, utxo_inputs_count(&inputs));
+  // test for -1 if transaction id is null
+  TEST_ASSERT(utxo_inputs_add(&inputs, 0, NULL, 2) == -1);
+
+  // test for -1 if index more than max_output_count
+  TEST_ASSERT(utxo_inputs_add(&inputs, 0, tx_id0, UINT8_MAX) == -1);
+
+  // test for -1 if index is 0
+  TEST_ASSERT(utxo_inputs_add(&inputs, 0, tx_id0, 0) == -1);
```

Comment:
index 0 is valid. see the comment above.

Reply:
Done.